src/main/kotlin/org/dif/crypto/key/SenderKeySelector.kt

```diff
@@ -1,23 +1,80 @@
 package org.dif.crypto.key
 
+import com.nimbusds.jose.jwk.Curve
 import org.dif.diddoc.DIDDocResolver
 import org.dif.exceptions.DIDDocException
 import org.dif.exceptions.DIDDocNotResolvedException
+import org.dif.exceptions.IncompatibleCryptoException
 import org.dif.exceptions.SecretNotFoundException
 import org.dif.secret.SecretResolver
+import org.dif.utils.divideDIDFragment
 import org.dif.utils.isDIDFragment
 
 class SenderKeySelector(private val didDocResolver: DIDDocResolver, private val secretResolver: SecretResolver) {
-    fun signKey(selector: String): Key = Key.wrapSecret(
-        if (isDIDFragment(selector)) {
-            secretResolver.findKey(selector).orElseThrow { throw SecretNotFoundException(selector) }
+    fun findSigningKey(signFrom: String): Key = Key.wrapSecret(
+        if (isDIDFragment(signFrom)) {
+            secretResolver.findKey(signFrom).orElseThrow { throw SecretNotFoundException(signFrom) }
         } else {
-            val didDoc = didDocResolver.resolve(selector).orElseThrow { throw DIDDocNotResolvedException(selector) }
+            val didDoc = didDocResolver.resolve(signFrom).orElseThrow { throw DIDDocNotResolvedException(signFrom) }
 
             val authentication = didDoc.authentications.firstOrNull()
-                ?: throw DIDDocException("Authentication is not found in DID Doc '$selector'")
+                ?: throw DIDDocException("The DID Doc '${didDoc.did}' does not contain compatible 'authentication' verification methods")
 
-            secretResolver.findKey(authentication).orElseThrow { throw SecretNotFoundException(selector) }
+            secretResolver.findKey(authentication).orElseThrow { throw SecretNotFoundException(signFrom) }
         }
     )
+
+    fun findAuthCryptKeys(from: String, to: String): Pair<Key, List<Key>> {
+        val (did) = divideDIDFragment(from)
+        val didDoc = didDocResolver.resolve(did).orElseThrow { throw DIDDocNotResolvedException(did) }
+
+        return if (isDIDFragment(from)) {
+            val sender = secretResolver.findKey(from)
+                .map { Key.wrapSecret(it) }
+                .orElseThrow { throw SecretNotFoundException(from) }
+
+            val recipients = findRecipientKeys(to, sender.curve)
+            Pair(sender, recipients)
+        } else {
+            didDoc.keyAgreements
+                .asSequence()
+                .map { secretResolver.findKey(it) }
+                .filter { it.isPresent }
+                .map { it.get() }
+                .map { Key.wrapSecret(it) }
+                .map { Pair(it, findRecipientKeys(to, it.curve)) }
+                .ifEmpty { throw DIDDocException("The DID Doc '${didDoc.did}' does not contain compatible 'keyAgreement' verification methods") }
```

Comment:
I think it should be `IncompatibleCryptoException`

Reply:
Not sure about it, in my opinion we should throw IncompatibleCryptoException if curves doesn't match